docs/installation/database/passing-configuration.rst

```diff
@@ -0,0 +1,430 @@
+.. _`choose one`: #The-Active-Environment
+.. _`environment variables`: #Generic-Environment-Variable-Overrides
+.. _`migration notes`: #Migrating-from-datacube-1.8
+
+1. Explicit configuration
+-------------------------
+
+Configuration can be passed in explicitly, without ever reading from a configuration file on disk.
+
+When explicit configuration is passed in, it takes precedence over configuration by environment variable.
+
+1a. Via Python (str or dict)
+++++++++++++++++++++++++++++
+
+A valid configuration dictionary can be passed in directly to the
+:py:class:`Datacube` constructor with the ``raw_config`` argument, without
+serialising to a string:
+
+.. code-block:: python
+
+   dc = Datacube(raw_config={
+      "default": {
+         "index_driver": "postgres",
+         "db_url": "postgresql:///mydb"
+      }
+   })
+
+The ``raw_config`` argument can also be passed config as a string, in either INI or YAML format:
+
+.. code-block:: python
+
+   dc = Datacube(raw_config="""
+   default:
+     # Connect to database mydb over local socket with OS authentication.
+     db_url: postgresql:///mydb
+   """)
+
+1b. As a string, via the datacube CLI
++++++++++++++++++++++++++++++++++++++
+
+The contents of a configuration file can be passed into the ``datacube`` CLI via the ``-R`` or
+``--raw-config`` command line option:
+
+::
+
+   datacube --raw-config "default: {db_database: this_db}"
+
+Output from a script that generates a configuration file dynamically can be passed in using
+a BASH backquote string:
+
+::
+
+   datacube --raw-config "`config_file_generator --option blah`"
+
+1c. As a string, via an Environment Variable
+++++++++++++++++++++++++++++++++++++++++++++
+
+If raw configuration has not been passed in explicitly via methods 1a. or 1b.
+above, then the contents of a configuration file can be written in full to the
+:envvar:`ODC_CONFIG` environment variable:
+
+.. code-block:: console
+
+   $ ODC_CONFIG="default: {db_database: this_db}"
+   $ datacube check    # will use the this_db database
+
+
+2. File configuration
+---------------------
+
+.. highlight:: python
+
+If explicit configuration has not been passed in, ODC attempts to find a configuration file.
+
+Only one configuration file is read.
+
+This is different to previous versions of the Open Data Cube,
+where multiple configuration files could be merged.
+
+If your previous practice was to extend a shared system configuration file with a local
+user configuration file, then you will now need to take a copy of the system configuration file,
+add your extensions to your copy, and ensure that the Open Data Cube reads from your
+modified file.
+
+2a. In Python
++++++++++++++
+
+In Python, the ``config`` argument can take a path to a config file:
+
+::
+
+    dc = Datacube(config="/path/to/my/file.conf")
+
+The ``config`` argument can also take a priority list of config paths.
+The first path in the list that can be read (i.e. exists and has read permissions) is read.
```

Comment:
How important is the information within the parenthesis, can it be omitted?

I think the sentence gets slightly easier to read if the last "is read" is replaced by "is used".